chunky/src/java/se/llbit/chunky/world/Chunk.java

```diff
@@ -421,8 +430,9 @@ public synchronized void renderTopography() {
    * @param palette Block palette
    * @param minY The requested minimum Y to be loaded into the chunkData object. The chunk implementation does NOT have to respect it
    * @param maxY The requested maximum Y to be loaded into the chunkData object. The chunk implementation does NOT have to respect it
+   * @throws ChunkLoadingException If there is an issue loading the chunk, and it should be aborted
    */
-  public synchronized void getChunkData(@NotNull Mutable<ChunkData> reuseChunkData, BlockPalette palette, BiomePalette biomePalette, int minY, int maxY) {
+  public synchronized void getChunkData(@NotNull Mutable<ChunkData> reuseChunkData, BlockPalette palette, BiomePalette biomePalette, int minY, int maxY) throws ChunkLoadingException {
```

Comment:
Should this be an `Exception` instead of a `RuntimeException`, since we are able to declare it everywhere?

Reply:
The reason it doesn't extend `Exception` is because non-`RuntimeException`s can't be thrown from `Executor#submit(Runnable)`, but they can be from `Executor#submit(Callable<T>)` where T is the return type.

Which is just bizarre. So we either extend `RuntimeException`, or we use a `Callable<Void>` and return `null`. I just thought it was a little janky